### PR TITLE
feat: xs validator option

### LINKED ISSUE #2676: how should validators choose to run vats on XS?
## What is the Problem Being Solved?

We currently use a `WORKER_TYPE=xs-worker` environment variable to test XS vats.

~~This is gross.~~ _Excuse me._ This is a risky use of ambient authority.

## Description of the Design

_something something genesis parameter something_?

_Feel free to re-assign back to me and/or @warner  when the design part is filled in._

## Security Considerations

?

## Test Plan

?

### PR DESCRIPTION
**Pull request by dckc**
fixes #2676

includes / depends on #2758

### CODE REVIEW

**Review by FUDCo — APPROVED** [review]
Is good

File: packages/cosmic-swingset/lib/ag-solo/start.js [hunk 1]
```diff
@@ -277,6 +281,10 @@ export default async function start(basedir, argv) {
     }
   }
 
+  const { wallet, defaultManagerType } = JSON.parse(
```
Comment (warner) [inline packages/cosmic-swingset/lib/ag-solo/start.js:284]: I don't know how this `options.json` works.. I want to make sure we aren't encouraging validator operators to think they can switch the type after creating their node. I'm a little suprised to think there are multiple places to set this value.

File: packages/SwingSet/src/initializeSwingset.js [hunk 2]
```diff
@@ -269,6 +270,23 @@ export async function initializeSwingset(
     config.devices = {};
   }
 
+  // Use ambient process.env only if caller did not specify.
+  const { env: { WORKER_TYPE } = process.env } = runtimeOptions;
+  const defaultManagerType = config.defaultManagerType || WORKER_TYPE;
```
Comment (warner) [inline packages/SwingSet/src/initializeSwingset.js]: Should we maybe name this `SWINGSET_WORKER_TYPE`? I wouldn't want it to collide with some other tooling that's talking about workers.

**Review by warner — APPROVED** [review]
Seems good.. consider renaming the env var, but it's not a biggie. I'm a little weirded out by the number of places where a default value is provided, but I imagine that helps avoid rewriting a lot of tests or something.

File: packages/cosmic-swingset/lib/ag-solo/start.js (hunk 1 above)
Comment (dckc) [reply packages/cosmic-swingset/lib/ag-solo/start.js:284]: `options.json` was the only path I could see between `make scenario3-setup` and `make scenario3-run-client`; i.e. between `ag-solo init` and `ag-solo start`.

Yes, it's not pretty, but I figure when in Rome...

File: packages/SwingSet/src/initializeSwingset.js (hunk 2 above)
Comment (dckc) [reply packages/SwingSet/src/initializeSwingset.js]: yes, that seems worthwhile.

File: packages/cosmic-swingset/lib/ag-solo/start.js (hunk 1 above)
Comment (warner) [reply packages/cosmic-swingset/lib/ag-solo/start.js:284]: @michaelfig does this seem good?

**Pull request merged**